File: Progetto_JAVA/src/progetto/Veicolo.java
package progetto;

//Classe Veicolo, superclasse di Automobile e Furgone
public class Veicolo {
	String proprietario;
	String targa;
	int lunghezza;
	int altezza;

	public Veicolo(String p, String t, int l, int a) {
		this.proprietario = p;
		this.targa = t;
		this.lunghezza = l;
		this.altezza = a;
	}

	public void printInfo() {
		System.out.println("Dati di " + this.targa + " :");
		System.out.print("Tipologia: Veicolo, ");
		System.out.print("Proprietario: " + this.proprietario);
		System.out.print(", Lunghezza: " + this.lunghezza);
		System.out.print(", Altezza: " + this.altezza);
	}

	public String getProprietario() {
		return proprietario;
	}

	public String getTarga() {
		return targa;
	}

	public int getLunghezza() {
		return lunghezza;
	}

	public int getAltezza() {
		return altezza;
	}

	
}

File: Progetto_C++_TESTA2/src/Veicoli/Veicolo.cpp
/*
 * Veicolo.cpp
 *
 *  Created on: 25 set 2023
 *      Author: Testa Lorenzo
 */

#include "Veicolo.h"
#include <iostream>
#include <sstream>
#include <string.h>
#include <vector>
#include <iterator>
using namespace std;


Veicolo::Veicolo(string p, string t, int l, int a) {
	this->Proprietario = p;
	this->Targa = t;
	this->Lunghezza = l;
	this->Altezza = a;
}

Veicolo::Veicolo(const Veicolo &other) {
	Proprietario = other.Proprietario;
	Targa = other.Targa;
	Lunghezza = other.Lunghezza;
	Altezza = other.Altezza;
}

inline const string Veicolo::getProprietario() {
	return Proprietario;
}
inline const string Veicolo::getTarga() {
	return Targa;
}
inline const int Veicolo::getLunghezza() {
	return Lunghezza;
}
inline const int Veicolo::getAltezza() {
	return Altezza;
}

const string Veicolo::getTOString() {
	stringstream streamer;
	streamer << "Dati di: Targa: " << this->Targa << " Proprietario: "
			<< this->Proprietario << endl << " Lunghezza: " << this->Lunghezza
			<< " Altezza: " << this->Altezza << endl;
	return streamer.str();
}

Veicolo::~Veicolo() {
}
